types/src/auth.rs

```diff
@@ -0,0 +1,157 @@
+//! types related to accounts
+
+#[cfg(feature = "tonic")]
+use pbjson_types::Any;
+use prost::Message;
+#[cfg(not(feature = "tonic"))]
+use prost_types::Any;
+
+use celestia_proto::cosmos::crypto::ed25519::PubKey as Ed25519PubKey;
+use celestia_proto::cosmos::crypto::secp256k1::PubKey as Secp256k1PubKey;
+use celestia_tendermint::public_key::PublicKey;
+use celestia_tendermint_proto::Protobuf;
+
+use crate::state::Address;
+use crate::Error;
+
+pub use celestia_proto::cosmos::auth::v1beta1::BaseAccount as RawBaseAccount;
+pub use celestia_proto::cosmos::auth::v1beta1::ModuleAccount as RawModuleAccount;
+
+const COSMOS_ED25519_PUBKEY: &str = "/cosmos.crypto.ed25519.PubKey";
+const COSMOS_SECP256K1_PUBKEY: &str = "/cosmos.crypto.secp256k1.PubKey";
+
+/// Params defines the parameters for the auth module.
+#[derive(Debug)]
+pub struct AuthParams {
```

Comment:
couldn't we just re-export `Params` as `AuthParams`? or is prost type tricky to work with?

Reply:
In my fury of adding types from protobufs I didn't notice we might as well use that one directly, switched :+1: